### ISSUE TITLE
Update selection queue after batch renaming files

### ISSUE THREAD

**Issue by Brixy** [sxyazi/yazi#731]
### What system are you running Yazi on?

Linux Wayland

### What terminal are you running Yazi in?

foot version: 1.16.2

### Yazi version

yazi 0.2.3 (d8abb3e 2024-02-26)

### Did you try the latest main branch to see if the problem got fixed?

Tried, but the problem still

### Describe the bug

Hi,

When you select files and batch rename them using `r`, the selection queue is not updated. The selection queue keeps the old names although these files don’t exist any more. The new/renamed files are not selected.

### Expected Behavior

I would expect the selection queue to be updated. i.e. the renamed/new files should be selected and the selection indicator be visible.

### To Reproduce

- Select N files.
- `r`.
- Rename some or all files.
- Selection counter shows that still N files are selected.
- `r` -> original file names are shown.

### Configuration

Yazi’s defaults.

**Comment by sxyazi** [comment]
Hi, please try this PR https://github.com/sxyazi/yazi/pull/791

**Comment by Brixy** [comment]
Works perfectly! Thank you very much.

Also, `unyank` works.

**Comment by Brixy** [comment]
Your PR seems to have solved another issue:

When you `yp`ed a directory (e.g. `yazi/` -> `yazi_1/`), entered, left and then deleted `yazi_1/`, then and used `p` again, `yazi_1/` _occasionally_ was empty (tested with `ls`), but yazi would shown the original files in the folder. I happened to notice this because I used this workflow for quick’n dirty backups. Anyway, it’s not happend after using your PR.
